### PR TITLE
Nonprivileged inline action subjective limit - develop

### PR DESCRIPTION
**Pull request by ndcgundlach**
## Change Description

Merge in changes from PR #9262 into develop.

Add new config file option max-nonprivileged-inline-action-size which is the cap for the size of an inline action above which a transaction will subjectively fail.

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by heifner** [comment]
Can we copy of the description for these types of PRs please. This makes for very ugly release notes.

**Comment by ndcgundlach** [comment]
> Can we copy over the description for these types of PRs please. This makes for very ugly release notes.

done

**Comment by heifner** [comment]
> done

And title. I personally like the release/2.0.x and develop to have the same title plus a `- develop` or `[2.0.x]`

**Comment by ndcgundlach** [comment]
> > done
> 
> And title. I personally like the release/2.0.x and develop to have the same title plus a `- develop` or `[2.0.x]`

Ok.

**Comment by ndcgundlach** [comment]
> Can the title be more descriptive please

Changed as per Kevin's suggestion.

File: unittests/test-contracts/test_api/test_api.cpp [hunk 1]
```diff
@@ -160,7 +162,7 @@ extern "C" {
       WASM_TEST_HANDLER_EX( test_permission, test_account_creation_time );
 
       //unhandled test call
-      eosio_assert( false, "Unknown Test" );
+      eosio_assert( false, "Unknown Test ahhh!" );
```
Comment (jgiszczak) [inline unittests/test-contracts/test_api/test_api.cpp:165]: Still necessary?

File: unittests/api_tests.cpp [hunk 2]
```diff
@@ -1203,6 +1211,132 @@ BOOST_FIXTURE_TEST_CASE(transaction_tests, TESTER) { try {
...
+   chain.produce_blocks(2);
+   chain.create_accounts( {N(testapi), N(testapi2), N(alice)} );
+   chain.set_code( N(testapi), contracts::test_api_wasm() );
+   chain.set_code( N(testapi2), contracts::test_api_wasm() );
+   chain.produce_block();
+
+   transaction_trace_ptr trace;
+   auto c = chain.control->applied_transaction.connect([&](std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> x) {
+         auto& t = std::get<0>(x);
+         if (t->scheduled) { trace = t; }
```
Comment (jgiszczak) [inline unittests/api_tests.cpp:1279]: How pervasive are the deferred transactions in test_api.wasm?  We should probably be preferring inline actions in new tests if possible without rewriting all tests immediately.

File: plugins/chain_plugin/chain_plugin.cpp [hunk 3]
```diff
@@ -332,6 +332,8 @@ void chain_plugin::set_program_options(options_description& cli, options_descrip
          }), "Number of threads to use for EOS VM OC tier-up")
          ("eos-vm-oc-enable", bpo::bool_switch(), "Enable EOS VM OC tier-up runtime")
 #endif
+         ("enable-account-queries", bpo::value<bool>()->default_value(false), "enable queries to find accounts by various metadata.")
```
[the remark this answers is not in the document]
Comment (ndcgundlach) [reply plugins/chain_plugin/chain_plugin.cpp]: removed

**Pull request merged**